src/MainWindow.vala

```diff
@@ -271,6 +274,46 @@ public class Tasks.MainWindow : Hdy.ApplicationWindow {
         });
     }
 
+    private void dialog_refresh_collection_error (Error e) {
+        string error_message = e.message;
+
+        GLib.Idle.add (() => {
```

Comment:
What's the reason for wrapping this in an idle?

Reply:
I (wrongly) assumed `GLib.Idle` is needed when we want to display something in the UI and this UI work is called from an async operation.